errbot/backends/irc.py

```diff
@@ -416,6 +430,21 @@ def on_dcc_connect(self, dcc, event):
     def on_dcc_disconnect(self, dcc, event):
         self.transfers.pop(dcc)
 
+    def on_endofnames(self, c, e):
+        # The e.arguments[0] contains the channel name.
+        # We filter that to avoid a misfire of the event.
+        room_name = e.arguments[0]
+        if room_name in self.__recently_joined_to:
```

Comment:
@zoni To avoid an Unnecessary thread locking, as this _recently_joined_to variable is Thread local variable, what do you think if i use [threading.local()](https://docs.python.org/3/library/threading.html#thread-local-data) and put this set into a the thread local variable context?